power_grid_model_c/power_grid_model/include/power_grid_model/optimizer/tap_position_optimizer.hpp

```diff
@@ -912,8 +914,7 @@ class TapPositionOptimizerImpl<std::tuple<TransformerTypes...>, StateCalculator,
         auto result = calculate_(state, method);
         ++total_iterations;
 
-        std::vector<IntS> iterations_per_rank(static_cast<signed char>(regulator_order.size() + 1),
-                                              static_cast<IntS>(0));
+        std::vector<IntS> iterations_per_rank(regulator_order.size());
```

Comment:
i see another problem: below, you do `            iterations_per_rank = rank_iterator.iterations_per_rank();`

but it should be

```
const auto& all_rank_iterations = rank_iterator.iterations_per_rank();
```

otherwise, you do a full assignment